src/main/java/wolox/training/models/Users.java

```diff
@@ -0,0 +1,99 @@
+package wolox.training.models;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import wolox.training.exceptions.BookAlreadyOwnedException;
+
+@Entity
+public class Users {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE)
+    private Long userId;
+
+    @Column(nullable = false)
+    private String username;
+
+    @Column(nullable = false)
+    private String name;
+
+    @Column(nullable = false)
+    private LocalDate birthdate;
+
+    @Column(nullable = false)
+    @ManyToMany
+    @JoinTable(name = "book_user",
+            joinColumns = @JoinColumn(name = "userId", referencedColumnName = "userId"),
+            inverseJoinColumns = @JoinColumn(name = "bookId",
+                    referencedColumnName = "bookId"))
+    private List<Book> books = new ArrayList<>();
+
+    public Users() {
+
+    }
+
+    public Long getUserId() {
+        return userId;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public void setUsername(String username) {
+        this.username = username;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public LocalDate getBirthdate() {
+        return birthdate;
+    }
+
+    public void setBirthdate(LocalDate birthdate) {
+        this.birthdate = birthdate;
+    }
+
+    public List<Book> getBooks() {
+        return (List<Book>) Collections.unmodifiableList(books);
+    }
+
+    public void setBooks(List<Book> books) {
+        this.books = books;
+    }
+
+    public void addBook(Book book) {
+
+        if (books.contains(book)) {
+            throw new BookAlreadyOwnedException();
+        }
+
+        books.add(book);
+    }
+
+    public void deleteBook(Long bookId) {
+        books.remove(filterBookById(bookId));
+    }
+
+    public Book filterBookById(Long bookId) {
+        return books.stream().filter(book -> (bookId).equals(book.getBookId())).collect(Collectors.toList()).get(0);
```

Comment:
Suggestion:
I recommend this code:

return books.stream()
            .filter(book -> bookId.equals(book.getBookId()))
            .findFirst()
            .orElseThrow(IllegalArgumentException::new - (other excepction));